### PR TITLE
feat: Update NodeChrome support latest version from GoogleChromeLabs

### LINKED ISSUE #2020: [🐛 Bug]: Could not start a new session. Response code 500. Message: Failed to read marionette port
### What happened?

When using any of the Firefox images and a random user ID (not the `seluser` ID in Ubuntu), session creation fails and GeckoDriver responds with "Failed to read marionette port".

### Command used to start Selenium Grid with Docker

```shell
docker run --user 24397 -d -p 4444:4444 --shm-size="2g" selenium/standalone-firefox:4.15.0-20231122
```

### Relevant log output

```shell
Could not start a new session. Response code 500. Message: Failed to read marionette port
```

### Operating System

Any of them

### Docker Selenium version (tag)

4.15.0-20231122

### PR DESCRIPTION
**Pull request by VietND96**
**Thanks for contributing to the Docker-Selenium project!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://selenium.dev/documentation/en/contributing/) guidelines, applied for this repository.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Motivation and Context
1. [GoogleChromeLabs](https://github.com/GoogleChromeLabs/chrome-for-testing) support a dashboard for versions https://googlechromelabs.github.io/chrome-for-testing
  NodeChrome image can get Chrome version based on ARG `CHROME_VERSION` with below inputs
```bash
       LATEST_RELEASE_STABLE
       LATEST_RELEASE_BETA
       LATEST_RELEASE_DEV
       LATEST_RELEASE_CANARY
       121 (e.g specific major version)
       119.0.6045.105 (e.g specific full version)
```
  By default, ARG `CHROME_DRIVER_VERSION` follows `CHROME_VERSION`. Chrome Driver version can be specified independently with the same support inputs above.

2. Add feat #2008
 [Reverted as duplicate with #2017]
 
3. After setting java prop `-Dwebdriver.chrome.driver` for NodeChrome. The below issue could be seen always.
Note: If removing this prop in the command start node, the issue is over, still not figure out reason for this strange
```bash
selenium.common.exceptions.SessionNotCreatedException: Message: Could not start a new session. Could not start a new session. Error while creating session with the driver service. Stopping driver service: Could not start a new session. Response code 500. Message: session not created: Chrome failed to start: exited normally.
  (session not created: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /opt/google/chrome/chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.) 
```
**Updated**
The above issue can be fixed by adding to node stereotype with the binary location, get from #2017
```{"browserName": "chrome", "browserVersion": "119.0", "platformName": "Linux", "goog:chromeOptions": {"binary": "/usr/bin/google-chrome"}}```
Change in test has been reverted.

4. Add test for helm chart, which covers #1994

5. Fix Firefox WebDriver issue when run with random user (true) #2020. Add back test case in workflow

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://selenium.dev/documentation/en/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/SeleniumTests/__init__.py [hunk 1]
```diff
@@ -68,8 +68,10 @@ def tearDown(self):
 
 class ChromeTests(SeleniumGenericTests):
     def setUp(self):
+        options=ChromeOptions()
+        options.add_argument('--no-sandbox')
```
Comment (giuliohome) [inline tests/SeleniumTests/__init__.py]: > such a configuration is unsupported and highly discouraged

https://chromedriver.chromium.org/help/chrome-doesn-t-start

Comment (VietND96) [reply tests/SeleniumTests/__init__.py]: Thanks for your review. This change in test has been reverted, since there is a fix from #2017

**Comment by VietND96** [comment]
Hi @diemol, I have also merged your updates in #2017 into this.

File: NodeChrome/Dockerfile [hunk 2]
```diff
@@ -35,27 +64,39 @@ USER 1200
...
   && sudo chmod 755 /opt/selenium/chromedriver-$CHROME_DRIVER_VERSION \
   && sudo ln -fs /opt/selenium/chromedriver-$CHROME_DRIVER_VERSION /usr/bin/chromedriver
+ENV SE_DRIVER_OPTS=" -Dwebdriver.chrome.driver="${CHROME_DRIVER_PATH_PROPERTY:-"/usr/bin/chromedriver"}" "$SE_DRIVER_OPTS
 
 #============================================
-# Dumping Browser name and version for config
+# Dumping Browser information for config
 #============================================
 RUN echo "chrome" > /opt/selenium/browser_name
+RUN google-chrome --version | awk '{print $5}' > /opt/selenium/browser_version
```
Comment (VietND96) [inline NodeChrome/Dockerfile:100]: Switch to install Chrome from https://googlechromelabs.github.io/chrome-for-testing
The response changed to `Google Chrome for Testing 119.0.6045.105 custom`
Use `$5` to get the version, instead of `$2` before

File: NodeBase/start-selenium-node.sh [hunk 3]
```diff
@@ -31,6 +31,10 @@ if [ ! -z "$SE_OPTS" ]; then
   echo "Appending Selenium options: ${SE_OPTS}"
 fi
 
+if [ ! -z "$SE_DRIVER_OPTS" ]; then
+  echo "Appending Selenium Driver Service property: ${SE_DRIVER_OPTS}"
+fi
+
```
Comment (diemol) [inline NodeBase/start-selenium-node.sh]: Please revert this change, we always want to set the browser driver property to avoid Selenium Manager in these images.

File: NodeBase/start-selenium-node.sh [hunk 4]
```diff
@@ -52,14 +56,8 @@ echo "Selenium Grid Node configuration: "
...
 echo "Starting Selenium Grid Node..."
 
-CHROME_DRIVER_PATH_PROPERTY=-Dwebdriver.chrome.driver=/usr/bin/chromedriver
-EDGE_DRIVER_PATH_PROPERTY=-Dwebdriver.edge.driver=/usr/bin/msedgedriver
-GECKO_DRIVER_PATH_PROPERTY=-Dwebdriver.gecko.driver=/usr/bin/geckodriver
-
 java ${JAVA_OPTS:-$SE_JAVA_OPTS} \
-  ${CHROME_DRIVER_PATH_PROPERTY} \
-  ${EDGE_DRIVER_PATH_PROPERTY} \
-  ${GECKO_DRIVER_PATH_PROPERTY} \
```
Comment (diemol) [inline NodeBase/start-selenium-node.sh]: Please leave it as it is currently. Same reasoning from the comment above.

File: NodeDocker/start-selenium-grid-docker.sh [hunk 5]
```diff
@@ -24,6 +24,10 @@ if [ ! -z "$SE_OPTS" ]; then
   echo "Appending Selenium options: ${SE_OPTS}"
 fi
 
+if [ ! -z "$SE_DRIVER_OPTS" ]; then
+  echo "Appending Selenium Driver Service property: ${SE_DRIVER_OPTS}"
+fi
+
```
Comment (diemol) [inline NodeDocker/start-selenium-grid-docker.sh]: This is not needed here, as the drivers are not used when using Dynamic Grid.

File: NodeDocker/start-selenium-grid-docker.sh [hunk 6]
```diff
@@ -42,6 +46,7 @@ else
 fi
 
 java ${JAVA_OPTS:-$SE_JAVA_OPTS} \
+  ${SE_DRIVER_OPTS} \
```
Comment (diemol) [inline NodeDocker/start-selenium-grid-docker.sh]: This is not needed here, as the drivers are not used when using Dynamic Grid.

File: NodeEdge/Dockerfile [hunk 7]
```diff
@@ -46,7 +44,10 @@ RUN if [ -z "$EDGE_DRIVER_VERSION" ]; \
   && rm /tmp/msedgedriver_linux64.zip \
   && mv /opt/selenium/msedgedriver /opt/selenium/msedgedriver-$EDGE_DRIVER_VERSION \
   && chmod 755 /opt/selenium/msedgedriver-$EDGE_DRIVER_VERSION \
-  && sudo ln -fs /opt/selenium/msedgedriver-$EDGE_DRIVER_VERSION /usr/bin/msedgedriver
+  && ln -fs /opt/selenium/msedgedriver-$EDGE_DRIVER_VERSION /usr/bin/msedgedriver
+ENV SE_DRIVER_OPTS="-Dwebdriver.edge.driver="${EDGE_DRIVER_PATH_PROPERTY:-"/usr/bin/msedgedriver"}" "$SE_DRIVER_OPTS
```
Comment (diemol) [inline NodeEdge/Dockerfile]: This won't be used as we are always setting the browser driver before launching the Grid.

File: NodeFirefox/Dockerfile [hunk 8]
```diff
@@ -35,9 +35,20 @@ RUN GK_VERSION=$(if [ ${GECKODRIVER_VERSION:-latest} = "latest" ]; then echo "0.
   && mv /opt/geckodriver /opt/geckodriver-$GK_VERSION \
   && chmod 755 /opt/geckodriver-$GK_VERSION \
   && ln -fs /opt/geckodriver-$GK_VERSION /usr/bin/geckodriver
+ENV SE_DRIVER_OPTS="-Dwebdriver.gecko.driver="${GECKO_DRIVER_PATH_PROPERTY:-"/usr/bin/geckodriver"}" "$SE_DRIVER_OPTS
```
Comment (diemol) [inline NodeFirefox/Dockerfile]: This won't be used as we are always setting the browser driver before launching the Grid.

File: NodeFirefox/Dockerfile [hunk 9]
```diff
@@ -35,9 +35,20 @@ RUN GK_VERSION=$(if [ ${GECKODRIVER_VERSION:-latest} = "latest" ]; then echo "0.
...
+    && chmod 777 /home/seluser/.cache /home/seluser/.mozilla /.cache
 
 USER 1200
 
+RUN if ! whoami &> /dev/null; then \
+  if [ -w /tmp/passwd ]; then \
+    echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:/home/seluser:/tmp:/bin/bash" >> /tmp/passwd; \
+  fi \
+fi
+
```
Comment (diemol) [inline NodeFirefox/Dockerfile]: Nice!

File: Standalone/start-selenium-standalone.sh [hunk 10]
```diff
@@ -11,6 +11,10 @@ if [ ! -z "$SE_OPTS" ]; then
   echo "Appending Selenium options: ${SE_OPTS}"
 fi
 
+if [ ! -z "$SE_DRIVER_OPTS" ]; then
+  echo "Appending Selenium Driver Service property: ${SE_DRIVER_OPTS}"
+fi
+
```
Comment (diemol) [inline Standalone/start-selenium-standalone.sh]: This won't be used as we are always setting the browser driver before launching the Grid.

File: Standalone/start-selenium-standalone.sh [hunk 11]
```diff
@@ -29,17 +33,11 @@ else
...
 fi
 
-CHROME_DRIVER_PATH_PROPERTY=-Dwebdriver.chrome.driver=/usr/bin/chromedriver
-EDGE_DRIVER_PATH_PROPERTY=-Dwebdriver.edge.driver=/usr/bin/msedgedriver
-GECKO_DRIVER_PATH_PROPERTY=-Dwebdriver.gecko.driver=/usr/bin/geckodriver
-
 java ${JAVA_OPTS:-$SE_JAVA_OPTS} \
-  ${CHROME_DRIVER_PATH_PROPERTY} \
-  ${EDGE_DRIVER_PATH_PROPERTY} \
-  ${GECKO_DRIVER_PATH_PROPERTY} \
```
Comment (diemol) [inline Standalone/start-selenium-standalone.sh]: Please leave it the way it currently is.

**Review by diemol — CHANGES_REQUESTED** [review]
Thank you, @vietnd96!

Feature 2 is not needed as we want to set the browser driver always. 

Point 3 has been fixed by setting the browser path in the stereotype.

**Comment by VietND96** [comment]
Thanks for your review @diemol !
I have reverted feat 2, keeping it the same as #2017 only.

**Review by diemol — APPROVED** [review]
Thank you, @vietnd96!
